lib/shared/bucketing-assembly-script/assembly/bucketing/segmentation.ts

```diff
@@ -406,16 +416,20 @@ export function getFilterValues(filter: UserFilter): JSON.Value[] {
  * If value has a datatype, use one of the type checkers above (eg. checkStringFilter)
  * NOTE: The use of Number.isNaN is required over the global isNaN as the check it performs is more specific
  */
-function checkValueExists(value: JSON.Value | null): bool {
+function checkValueExists(value: CustomDataValue | null): bool {
     if (!value) return false
-    const stringValue = value.isString ? value as JSON.Str : null
-    const floatValue = value.isFloat ? value as JSON.Float : null
-    const intValue = value.isInteger ? value as JSON.Integer : null
-    const boolValue = value.isBool ? value as JSON.Bool : null
-
-    return value !== null
-        && !!(stringValue || floatValue || intValue || boolValue)
-        && (!stringValue || stringValue.valueOf() !== '')
-        && (!floatValue || !isNaN(floatValue.valueOf()))
-        && (!intValue || !isNaN(intValue.valueOf()))
+    if (CustomDataValueInterpreter.isString(value)) {
+        const stringValue = CustomDataValueInterpreter.asString(value)
+        return stringValue !== null && stringValue !== ''
+    }
+    if (CustomDataValueInterpreter.isFloat(value)) {
+        const floatValue = CustomDataValueInterpreter.asNumber(value)
+        return floatValue !== null && !isNaN(floatValue)
```

Comment:
I guess this means there are no more int values for customData values, I wonder how that affects things...

Reply:
I don't think there were anyway, since we were using protobuf to construct the user JSON object and protobuf didnt have ints